roles/config-rh-sso/tasks/add_role.yml

```diff
@@ -0,0 +1,30 @@
+---
+
+- name: "Requesting Access Token"
+  uri:
+    url: "{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/realms/master/protocol/openid-connect/token"
```

Comment:
I've noticed that in your roles you've been using `"{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/realms/master`. It actually looks like this means that it creates it directly into the `master` realm. We should add an item to the inventory for each role to specify the realm they want to create this role in. Then we should take that value and replace `master` with it.

example:

```
roles:
  - role_name: sample-role
    role_realm: a-simple-realm
    role_description: "Example of role created via an API call"
#   role_attributes:
#     - key1: value
    role_clientRole: false
    role_composite: false
```

Then in the task it would look something like:

``"{{ rh_sso_protocol }}://{{ rh_sso_host }}:{{ rh_sso_port }}/auth/realms/{{ role_data.realm }}``

Reply:
The only one that we don't need to change is when we're grabbing a token.

Reply:
Perfect. I added the target Role variable in the inventory and called it `role_targetRealm` that points to the target where we want our Role to be created in.